Add CVE justification for CVE-2025-46392 (#129)
wso2/docs-security: en/docs/security-announcements/cve-justifications/2025/CVE-2025-46392.md

---
title: "CVE-2025-46392"
category: security-announcements
date: 
---

# CVE-2025-46392

<p class="doc-info">WSO2 Products impacted: no</p>
<p class="doc-info">Customers actions required: no</p>
---

### REPORTED VULNERABILITY

> A Denial of Service attack through uncontrolled resource consumption by providing a crafted config file with malicious or recursive variable interpolation. [^1]

Reviewer: ```suggestion
A Denial of Service (DoS) vulnerability in Apache Commons Configuration 1.x which can be exploited by supplying a specially crafted configuration file containing malicious or recursive variable interpolation, leading to uncontrolled resource consumption.. [^1]
```

Author: Thanks for the suggestion. I do agree and I have committed the changes.

### REPORTED PRODUCTS

- WSO2 Identity Server : 7.0.0, 7.1.0

### WSO2 JUSTIFICATION

The reported vulnerability involves uncontrolled resource consumption via malicious or recursive variable interpolation in configuration files. In WSO2 Identity Server, the affected component is used exclusively to load the log4j2.properties file. This file is not user-controllable under normal deployment scenarios.

> Therefore, unless an attacker has write access to the server’s filesystem or configuration directories, the vulnerability is not exploitable in practice. Assuming proper server hardening and file system access controls, this issue poses no practical risk.

Author: Ack. Resolved with https://github.com/wso2/docs-security/pull/129/commits/8046141fbf588183211e54edb2ba62fc25b9d67b

---
title: "CVE-2025-46392"
category: security-announcements
date: July 31 2025
---

# CVE-2025-46392

<p class="doc-info">WSO2 Products impacted: no</p>
<p class="doc-info">Customers actions required: no</p>
---

### REPORTED VULNERABILITY

> A Denial of Service (DoS) vulnerability in Apache Commons Configuration 1.x which can be exploited by supplying a specially crafted configuration file containing malicious or recursive variable interpolation, leading to uncontrolled resource consumption.. [^1]

Reviewer 2: ```suggestion
A Denial of Service (DoS) vulnerability in Apache Commons Configuration 1.x which can be exploited by supplying a specially crafted configuration file containing malicious or recursive variable interpolation, leading to uncontrolled resource consumption [^1].
```

### REPORTED PRODUCTS

- WSO2 Identity Server : 7.0.0, 7.1.0

### WSO2 JUSTIFICATION

> Upon evaluating the impact of upgrading Apache Commons Configuration from version 1.x to 2.x, we identified significant architectural changes that introduce compatibility issues. Version 1.x uses the org.apache.commons.configuration.* package hierarchy, whereas version 2.x has been restructured under org.apache.commons.configuration2.*, with major modifications. Additionally, to make commons-configuration2 function as an OSGi service within the product, several new dependencies must be introduced, increasing the integration complexity.

Reviewer: ```suggestion
Upon evaluating the impact of upgrading Apache Commons Configuration from version 1.x to 2.x, we identified significant architectural changes that introduce compatibility issues. Version 1.x uses the org.apache.commons.configuration.* package hierarchy, whereas version 2.x has been restructured under org.apache.commons.configuration2.*, with major modifications. Additionally, to make commons-configuration2 function as an OSGi service within the product, several new dependencies must be introduced, increasing the integration complexity. Due to these concerns, we are publishing this CVE justification with a detailed analysis of the CVE, outlining how associated risks are mitigated in WSO2 products.
```

> Upon evaluating the impact of upgrading Apache Commons Configuration from version 1.x to 2.x, we identified significant architectural changes that introduce compatibility issues. Version 1.x uses the org.apache.commons.configuration.* package hierarchy, whereas version 2.x has been restructured under org.apache.commons.configuration2.*, with major modifications. Additionally, to make commons-configuration2 function as an OSGi service within the product, several new dependencies must be introduced, increasing the integration complexity. Due to these concerns, we are publishing this CVE justification with a detailed analysis of the CVE, outlining how associated risks are mitigated in WSO2 products.

Reviewer 2: Confused with `*` in the paragraph

> The reported vulnerability involves uncontrolled resource consumption via malicious or recursive variable interpolation in configuration files. In WSO2 Identity Server, the affected component is used exclusively to load the log4j2.properties file. This file is not user-controllable under normal deployment scenarios.

Reviewer 2: ```suggestion
The reported vulnerability involves uncontrolled resource consumption via malicious or recursive variable interpolation in configuration files. In WSO2 Identity Server, the affected component is used exclusively to load the `log4j2.properties` file. This file is not user-controllable under normal deployment scenarios.
```

> normal deployment scenarios.
Can there be any other scenarios? If not, shall we remove "normal" word from here?

> Therefore, unless an attacker has write access to the server’s filesystem or configuration directories, the vulnerability is not exploitable in practice. Assuming proper server hardening and file system access controls, this issue poses no practical risk.

Reviewer 3: @ayomawdb @JeethJJ Do we also need to say that we will address this in the upcoming release?

Author: My impression was that, since its a justification for the CVE, we do not necessarily need to mention that we will fix in later versions. We can add if necessary. Lets get the final call from @ayomawdb

Reviewer: Up to the product to decide this, but there must be a roadmap item for these in the product side to make sure we do track these in future. We will be highlighting these issues again in future pre-release scans and at that time also pointing to this CVE justification is not right. Hence moving from Apache Commons Configuration 1.x to Apache Commons Configuration 2.x should be a roadmap item for next major release, internally.

Reviewer: @JeethJJ @DMHP Please update once this done.

Author: Please find the roadmap item at [1]

[1] https://github.com/wso2/product-is/issues/24947

> 

Reviewer 4: In the other past CVE justifications, we had a section called CONCLUSION. May be it is good to maintain the same structure, WDYT?

Author: As the one I was referring to did not have a conclusion [1], seems I've missed it. Also it seems we have added both with and without CONCLUSION section. @ayomawdb WDYT?

[1] https://security.docs.wso2.com/en/latest/security-announcements/cve-justifications/2025/CVE-2025-47889/

Reviewer: Not having CONCLUSION section in this PR is fine. We will unify all CVE justifications from our end first related to CONCLUSION section and then enforce this requirement for future justifications,
